x-pack/plugins/observability/public/hooks/slo/use_fetch_slo_instances.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { GetSLOInstancesResponse } from '@kbn/slo-schema';
+import { useQuery } from '@tanstack/react-query';
+
+import { useKibana } from '../../utils/kibana_react';
+import { sloKeys } from './query_key_factory';
+
+export interface UseFetchSloInstancesResponse {
+  isInitialLoading: boolean;
+  isLoading: boolean;
+  isRefetching: boolean;
+  isSuccess: boolean;
+  isError: boolean;
+  data: GetSLOInstancesResponse | undefined;
+}
+
+export function useFetchSloInstances({ sloId }: { sloId?: string }): UseFetchSloInstancesResponse {
```

Comment:
💬 Currently not used, but could be very soon in the SLO details page, instance dropdown selector